### PR TITLE
Make call to compilation.CompleteTrees really unconditional.

### PR DESCRIPTION
**Pull request by VSadov**
Make call to compilation.CompleteTrees really unconditional.

Fixes https://github.com/dotnet/roslyn/issues/11368

### CODE REVIEW

**Comment by mavasani** [comment]
👍 Thanks!

File: src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs [hunk 1]
```diff
@@ -508,11 +508,11 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                                     {
                                         compilation.ReportUnusedImports(null, diagnosticBag, cancellationToken);
                                     }
-
-                                    compilation.CompleteTrees(null);
                                 }
                             }
 
+                            compilation.CompleteTrees(null);
```
Comment (jaredpar) [inline src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs]: In the case of an exception this method won't be called.  This meant the async queue won't follow it's desired state transition.  Will that cause deadlocks / issues down the road?  What are the guarantees we provide?  

@gafter may have input here.

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-compiler

**Comment by AlekseyTs** [comment]
LGTM

**Comment by VSadov** [comment]
@dotnet/roslyn-compiler  @mavasani  @gafter  Added another small fix and removed the SpinWait. 
I think, to the best of our knowledge, this should work reliably in GetDiagnostics scenario.

Added some relevant comments.

**Comment by VSadov** [comment]
I am going to check this in since the bug randomly breaks builds. 
The PR should be considered still open for comments.

**Comment by cston** [comment]
LGTM

**Pull request merged**